src/script.c

```diff
@@ -501,3 +504,36 @@ void InitRamScript_NoObjectEvent(u8 *script, u16 scriptSize)
     InitRamScript(script, scriptSize, MAP_GROUP(UNDEFINED), MAP_NUM(UNDEFINED), NO_OBJECT);
 #endif //FREE_MYSTERY_EVENT_BUFFERS
 }
+
+#define WALK_AWAY_SIGNPOST_FRAMES 6
+
+void SetWalkingIntoSignVars(void)
+{
+    gWalkAwayFromSignpostTimer = WALK_AWAY_SIGNPOST_FRAMES;
+    sMsgBoxIsCancelable = TRUE;
+}
+
+bool32 IsMsgSignPost(void)
+{
+    return sMsgIsSignPost;
+}
+
+void ResetFacingNpcOrSignPostVars(void)
+{
+    sMsgIsSignPost = FALSE;
+}
+
+void MsgSetSignPost(void)
+{
+    sMsgIsSignPost = TRUE;
+}
+
+void ClearMsgBoxCancelableState(void)
+{
+    sMsgBoxIsCancelable = FALSE;
+}
+
+bool32 CanWalkAwayToCancelMsgBox(void)
+{
+    return sMsgBoxIsCancelable;
+}
```

Comment:
These are all trivial enough that they don't need to be functions imo

Reply:
When I previously tried to do this, `sMsgIsSignPost` always returned `FALSE`.

Reply:
https://github.com/rh-hideout/pokeemerald-expansion/pull/5044/commits/846e8a29daadc96035944250dde9b391e8fe2846
https://github.com/rh-hideout/pokeemerald-expansion/pull/5044/commits/c3c433439e2f5745782b01ef42e6915783dd5655